### PR TITLE
[BOP-1278] Add release matrix of MKE 4 component versions

### PR DESCRIPTION
**Pull request by RaunakShah**
This PR introduces the MKE 4 release matrix in JSON format, as well as a JSON schema for the release matrix.
`mkectl` will pull this JSON and apply/update the component versions based on their entry in the matrix. 
There's also a `minimumMkectlVersion` that represents the version of `mkectl` that can manage a given MKE 4 release.

Also added a makefile target to validate that the json file matches the json schema.

### CODE REVIEW

File: release-matrix/release-matrix-schema.json [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      },
+      "mkeOperatorVersion": {
+        "type": "string",
+        "description": "MKE Operator version for this MKE 4 release"
+      },
+      "k0sVersion": {
+        "type": "string",
+        "description": "k0s version for this MKE 4 release"
+      },
+      "isLatest": {
```
Comment (vikramhh) [inline release-matrix/release-matrix-schema.json:25]: @RaunakShah - why is this needed?

File: release-matrix/release-matrix-schema.json [hunk 2]
```diff
@@ -0,0 +1,36 @@
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "title": "MKE 4 Release Matrix",
+  "description": "",
+  "type": "array",
+  "items": {
+    "type": "object",
+    "properties": {
+      "clusterVersion": {
```
Comment (tppolkow) [inline release-matrix/release-matrix-schema.json:9]: nit: perhaps this should be `release` ,  `releaseVersion` or `mkeReleaseVersion`? It wasn't immediately obvious what this version represented to me.

File: release-matrix/release-matrix-schema.json [hunk 3]
```diff
@@ -0,0 +1,36 @@
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "title": "MKE 4 Release Matrix",
```
Comment (tppolkow) [inline release-matrix/release-matrix-schema.json:3]: Suggest: Release Compatibility Matrix?

File: release-matrix/release-matrix-schema.json (hunk 1 above)
Comment (RaunakShah) [reply release-matrix/release-matrix-schema.json:25]: @vikramhh as discussed in the orca cafe this week and also prior meetings, we want to allow users to specify "latest" as their upgrade version. This is to identify the latest supported release version.

File: release-matrix/release-matrix-schema.json (hunk 3 above)
Comment (RaunakShah) [reply release-matrix/release-matrix-schema.json]: Done

File: release-matrix/release-matrix-schema.json (hunk 2 above)
Comment (RaunakShah) [reply release-matrix/release-matrix-schema.json]: I went with `mkeReleaseVersion` from your suggestions. I didn't want to cause confusion between `mkectlVersion`, `mkeOperatorVersion` and `mkeVersion` originally

**Review by tppolkow — APPROVED** [review]
> I couldn't find a way to enforce the schema via GHA or a shell script, so I've introduced a README to describe how to update this file for future releases.

I think you could likely do this via adding a Makefile target that
1) Build an image with `check-jsonschema` in it and mounts `release-matrix/` directory
2) Runs validation command
3) Check exit code of container

And a GHA that just calls the makefile target, like we do in `mke` repo via `run:`. 

I think this would be nice to have but the json is simple enough that I don't anticipate many issues with the json schema and we have the manual approach. I will approve PR without it and leave it up to you if you want to try that.

**Review by tppolkow — APPROVED** [review]
@RaunakShah Looks good!
